R/make_vaccineff_data.R

```diff
@@ -0,0 +1,254 @@
+    matching <- NULL
+
+    if (!is.null(t0_follow_up)) {
+      start_from_immunization <- TRUE
+    } else {
+      start_from_immunization <- FALSE
+    }
+
+    cohort_data$outcome_status <- set_event_status(
+      data_set = cohort_data,
+      outcome_date_col = outcome_date_col,
+      censoring_date_col = censoring_date_col
+    )
+
+    cohort_data$time_to_event <- get_time_to_event(
+      data_set = cohort_data,
+      outcome_date_col = outcome_date_col,
+      start_cohort = start_cohort,
+      end_cohort = end_cohort,
+      start_from_immunization = start_from_immunization,
+      immunization_date_col = t0_follow_up
+    )
+
+    output <- capture_warnings(adjust_time_to_event(
+      data_set = cohort_data
+    ))
+
+    cohort_data <- output$result
+    warnings_log <- output$warnings
+
+    cohort_data <- linelist::set_tags(
+      x = cohort_data,
+      time_to_event_col = "time_to_event",
+      outcome_status_col = "outcome_status",
+      allow_extra = TRUE
+    )
+  }
+
+  vaccineff_data <- list(
+    cohort_data = cohort_data,
+    start_cohort = start_cohort,
+    end_cohort = end_cohort,
+    vaccinated_status = vaccinated_status,
+    unvaccinated_status = unvaccinated_status,
+    immunization_delay = immunization_delay,
+    matching = matching,
+    warnings_log = warnings_log
+  )
+
+  class(vaccineff_data) <- "vaccineff_data"
+
+  return(vaccineff_data)
+}
+
+#' @title Summarize Vaccineff Data
+#'
+#' @description Summarizes the results of `make_vaccineff_data`.
+#'
+#' @param object Object of the class `vaccineff_data`.
+#' @param warnings_log If `TRUE`, prints the warnings log.
+#' @param ... Additional arguments passed to other functions.
+#' @return Summary of the results from vaccineff data
+#' @export
+
+summary.vaccineff_data <- function(object, warnings_log = FALSE, ...) {
+  # Check if the input object is of class "match"
+  stopifnot("Input must be an object of class 'vaccineff_data'" =
+      checkmate::test_class(object, "vaccineff_data")
+  )
+
+  cat(paste0("\nCohort start: ", object$start_cohort))
+  cat(paste0("\nCohort end: ", object$end_cohort, "\n"))
+
+  # Extract tags
+  tags <- linelist::tags(object$cohort_data)
+
+  # Summary
+  if (!is.null(object$matching)) {
+    cat("\nNearest neighbors matching iteratively performed.\n")
+    summary.match(object$matching)
+
+    if (warnings_log && !is.null(object$matching)) {
+      cat("\nWarnings:\n")
+      cat(object$warnings_log, sep = "")
+    }
+
+  } else {
+    summ_cohort <- match_summary(
+      all = object$cohort_data,
+      matched = NULL,
+      vacc_status_col = tags$vacc_status_col
+    )
+    cat("\nNo matching routine invoked.\n")
+    print(summ_cohort)
+
+    cat(object$warnings_log, sep = "")
+  }
+
+  # Print tags from linelist object
+  tags_txt <- paste(names(tags), unlist(tags), sep = ":", collapse = ", ")
+  if (tags_txt == "") {
+    tags_txt <- "[no tagged variable]"
+  }
+  cat("\n// tags:", tags_txt, "\n")
```

Comment:
I think you should explicitly use the argument `sep = "\n"` for it to act as a line break.

Reply:
Thanks for the suggestion! I use the line break here just at the end of the line. If I used it as a separator, it would create a lot of noise on the screen. I actually recycled this code from linelist, haha.